### ISSUE TITLE
Deno.realPath returns mangled network SMB path by prefixing it with UNC

### ISSUE THREAD

**Issue by midnightnoon** [denoland/deno#12243]
Deno.realPath and realPathSync mangles network share paths on Windows by prefixing them with 'UNC'.

For example, assuming `\\192.168.1.100\share\` is a valid network share:

```
> deno eval "console.log(await Deno.realPath('\\\\192.168.1.100\\share\\'))"
UNC\192.168.1.100\share
```

Subsequent calls to `Deno.readDir` with that path results in a ` The system cannot find the path specified. (os error 3)`  message. Manually replacing `UNC` with `\` returns the expected behaviour.

**Comment by midnightnoon** [comment]
This is on 1.14.1

**Comment by MarkTiedemann** [comment]
This is missing the extended path-length prefix `\\?\` (to avoid `MAX_PATH`), e.g. `\\?\UNC\$host\$share`.

**Comment by MarkTiedemann** [comment]
Looking at `canonicalize_path`, this prefixed is deliberately trimmed.

https://github.com/denoland/deno/blob/main/cli/fs_util.rs#L69

There should probably be an exception for shares.

**Comment by lucacasonato** [comment]
@piscisaureus can you comment on if this something that should be changed?

**Comment by MarkTiedemann** [comment]
I believe this bug is still present.

**Comment by piscisaureus** [comment]
I think this should have been fixed by https://github.com/denoland/deno/pull/12438.

**Comment by MarkTiedemann** [comment]
Oh, you're right. I was still running v1.15. It is fixed in v1.16.

Thanks!
